plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/GpeMigrator.java

```diff
@@ -108,29 +120,35 @@ private static boolean isGpeClasspath(IClasspathEntry entry) {
   }
 
   @VisibleForTesting
-  static void removeGpeNature(IProject project) {
+  static boolean removeGpeNature(IProject project) {
     try {
+      boolean hadNature = NatureUtils.hasNature(project, GPE_GAE_NATURE_ID);
       NatureUtils.removeNature(project, GPE_GAE_NATURE_ID);
+      return hadNature;
     } catch (CoreException ex) {
       logger.log(Level.WARNING, "Failed to remove GPE nature.", ex);
+      return true;
```

Comment:
I'm going to back out these changes to `GpeMigrator` method return-type as I didn't end up using them.